### PR TITLE
Desktop ready modal

### LINKED ISSUE #2825: [COMPONENT] Fullscreen Modal
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the solution you'd like

# Acceptance criteria  (AC): 
The fullscreen modal should have different behaviour and look, depending on if it is rendered on desktop or mobile.

- [ ] Desktop
  - [X] The modal animates in from the bottom
  - [X] The modal is centered on the screen
  - [X] The width of the modal can be configured with t-shirt sizes (S, M, L)
  - [X] When using t-shirt sizes, the modal automatically retains a certain aspect ratio (4:3?)
  - [X] The height can be set by the developer in pixels and not t-shirt sizes
  - [X] The modal has a padding that prevents it from touching the edges of the viewport
- [ ] Mobile
  - [X] The modal animates in from the bottom
  - [X] The modal appears as a fullscreen overlay
  - [ ] We have decide whether we use the card or the fullscreen modal - TBD

## Have you considered any alternatives?
No

## Are there any additional context?
No

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by mark-drastrup**
## Which issue does this PR close?

This PR closes #2825

## What is the new behavior?
The new desktop ready modal that can be used both as a component or instantiated through the modalController.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [X] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [X] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [X] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [X] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [X] Request that the changes are code-reviewed 
- [X] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/modal/experimental/src/modal/modal.component.ts [hunk 1]
```diff
@@ -3,6 +3,9 @@ import { IonContent, IonModal } from '@ionic/angular';
 import { OverlayEventDetail } from '@ionic/core/components';
 import { KirbyAnimation } from '@kirbydesign/designsystem/helpers';
 
+type Size = 'xxs' | 'xs' | 'sm' | 'md' | 'lg';
+type SizeTemp = 'md';
```
Comment (mark-drastrup) [inline libs/designsystem/modal/experimental/src/modal/modal.component.ts:7]: We will soon get the correct t-shirt sizes, but for now `md` is the default size.

Comment (mictro) [reply libs/designsystem/modal/experimental/src/modal/modal.component.ts:7]: ok

File: libs/designsystem/modal/experimental/src/services/modal.controller.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { from, Observable, Subject, switchMap, tap } from 'rxjs';
 import { OverlayEventDetail } from '@ionic/core/components';
 
 export type ModalFlavor = 'modal';
+type Size = 'md';
```
Comment (mictro) [inline libs/designsystem/modal/experimental/src/services/modal.controller.ts]: Same as above?

File: libs/designsystem/modal/experimental/src/services/modal.controller.ts [hunk 3]
```diff
@@ -34,21 +37,35 @@ export class ModalExperimentalController {
...
 
     this.isModalOpening = true;
 
     modal$
       .pipe(
+        tap((modal) => {
+          if (config.height) {
+            modal.style.setProperty('--height', config.height);
+          }
+        }),
```
Comment (mictro) [inline libs/designsystem/modal/experimental/src/services/modal.controller.ts]: tap med sideffekter?

File: libs/designsystem/modal/experimental/src/wrapper/wrapper.component.scss [hunk 4]
```diff
@@ -1,40 +1,62 @@
...
+  --footer-height: 0px;
+}
 
-    button {
-      color: var(--color);
-    }
+ion-modal {
+  --background: var(--kirby-modal-background, #{utils.get-color('background-color')});
+  --max-height: 928px;
+  --min-height: 594px;
```
Comment (mictro) [inline libs/designsystem/modal/experimental/src/wrapper/wrapper.component.scss]: benyt lokale scss variable.

File: libs/designsystem/modal/experimental/src/wrapper/wrapper.component.scss [hunk 5]
```diff
@@ -1,40 +1,62 @@
...
 
-    @include phablet-toolbar-padding;
+  @include utils.media('>=medium') {
+    --border-radius: 32px;
```
Comment (mictro) [inline libs/designsystem/modal/experimental/src/wrapper/wrapper.component.scss]: same as above...

File: libs/designsystem/modal/experimental/src/modal/modal.component.html [hunk 6]
```diff
@@ -1,18 +1,21 @@
 <ion-modal
+  class="kirby-modal-experimental"
   [isOpen]="open"
   [canDismiss]="canDismiss"
+  [ngClass]="size"
+  [style.--height]="height"
```
Comment (mictro) [inline libs/designsystem/modal/experimental/src/modal/modal.component.html]: A bit cryptic - can it be change to a more comprehensible ala 'height: (user-height ? user-height : --height)' perhaps using the model?

**Review by mictro — CHANGES_REQUESTED** [review]

**Review by mictro — APPROVED** [review]

**Comment by mictro** [comment]
ezgif-3-a9801f242c

**Pull request merged**